db/models/comments.model.js

```diff
@@ -1,13 +1,33 @@
+const { response } = require('../app')
 const db = require('../connection')
 
-function selectAllComments(article_id){
+function selectAllComments(article_id) {
     return db.query('SELECT * FROM comments WHERE article_id=$1 ORDER BY created_at DESC', [article_id])
-    .then((result) => {
-        if (result.rows.length === 0){
-            return Promise.reject({status: 404, msg: 'article does not exist'})
-        }
-        return result.rows
-    })
+        .then((result) => {
+            if (result.rows.length === 0) {
+                return Promise.reject({ status: 404, msg: 'article does not exist' })
+            }
+            return result.rows
+        })
 }
 
-module.exports = selectAllComments
\ No newline at end of file
+function insertComment(article_id, { author, body }) {
+    return db.query('INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *', [article_id, author, body])
+        .then((result) => {
+            if (result.rows[0].body.length === 0){
+                return Promise.reject({ status: 400, msg: 'Bad request'})
+            }
+            return result.rows[0].body
+        })
+        .catch((err) => {
+            if (err.detail === 'Key (article_id)=(90) is not present in table "articles".') {
```

Comment:
This is very brittle and won't work if any other non-existent article id is passed. However, you have the right idea about looking at the error message. 

Article_id is a foreign key so it will cause a PSQL error to be flagged -instead of err.detail You should just get the PSQL error code (err.code), look it up on google or in the [documentation](https://www.postgresql.org/docs/current/errcodes-appendix.html) to see if it's the error code you were expecting and use it to make a PSQL error handler [as shown here](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)